spec/puppetserver/ca/action/prune_spec.rb

```diff
@@ -183,6 +224,105 @@
+      expect(number_of_removed_crl_entries).to eq(2)
+      expect(ca_crl.revoked.length).to eq(2)
+    end
+
+    it 'continue to scan CA directory and remove expired entries if inventory.txt is missing' do
+      allow(connection).to receive(:get).and_return(offline)
+      ca_key = OpenSSL::PKey::RSA.new(512)
+      ca_cert = create_cert(ca_key, "You-Shall-Not-Pass")
+      revoked_cert = create_cert(ca_key, 'foo')
+      revoked_cert.not_before = Time.now - (60 * 60 * 24 * 365 * 2)
+      revoked_cert.not_after = Time.now  - (60 * 60 * 24 * 365 * 1)
+      ca_crl = create_crl(ca_cert, ca_key, [revoked_cert])
+      inventory_entries = ca.inventory_entry(revoked_cert)
+      File.write("#{settings[:cadir]}/signed/foo.pem", revoked_cert)
+      allow(File).to receive(:exist?).with(settings[:cert_inventory]).and_return(false)
+      allow(File).to receive(:exist?).with("#{settings[:cadir]}/signed/foo.pem").and_return(true)
+      number_of_removed_crl_entries = subject.prune_expired(ca_crl, ca_key, settings[:cert_inventory], settings[:cadir])
+      expect(number_of_removed_crl_entries).to eq(1)
+      expect(ca_crl.revoked.length).to eq(0)
+    end
+
+    it 'correctly parse inventory.txt file with miscellaneous entries when checking for expired entries' do
+      allow(connection).to receive(:get).and_return(offline)
+      ca_key = OpenSSL::PKey::RSA.new(512)
+      ca_cert = create_cert(ca_key, "You-Shall-Not-Pass")
+      revoked_cert = create_cert(ca_key, 'foo.example.com')
+      revoked_cert.not_before = Time.now - (60 * 60 * 24 * 365 * 2)
+      revoked_cert.not_after = Time.now  - 1
+      ca_crl = create_crl(ca_cert, ca_key, Array.new(1, revoked_cert))
+      File.write(settings[:cert_inventory], "\n"+"I am bad foo.example.com\n"+ca.inventory_entry(revoked_cert)+"\nHello")
+      allow(File).to receive(:exist?).and_return(true)
+      number_of_removed_crl_entries = subject.prune_expired(ca_crl, ca_key, settings[:cert_inventory], settings[:cadir])
+      expect(number_of_removed_crl_entries).to eq(1)
+      expect(ca_crl.revoked.length).to eq(0)
+    end
+
+    it 'scan CA directory and remove expired entries even if miscellaneous files exist' do
+      allow(connection).to receive(:get).and_return(offline)
+      ca_key = OpenSSL::PKey::RSA.new(512)
+      ca_cert = create_cert(ca_key, "You-Shall-Not-Pass")
+      revoked_cert = create_cert(ca_key, 'foo')
+      revoked_cert.not_before = Time.now - (60 * 60 * 24 * 365 * 2)
+      revoked_cert.not_after = Time.now  - (60 * 60 * 24)
+      ca_crl = create_crl(ca_cert, ca_key, [revoked_cert])
+      inventory_entries = ca.inventory_entry(revoked_cert)
+      File.write("#{settings[:cadir]}/signed/foo.pem", revoked_cert)
+      File.write("#{settings[:cadir]}/signed/hello", "hello")
+      File.write("#{settings[:cadir]}/signed/test.txt", "testing")
+      allow(File).to receive(:exist?).with(settings[:cert_inventory]).and_return(false)
```

Comment:
Instead of stubbing out a bunch of this stuff, perhaps we can actually write the files?

Reply:
In that case the puppet config needs to be written. So I am going to leave it as it is.